### ISSUE TITLE
Retry causes unrecoverable error

### ISSUE THREAD

**Issue by 89erik** [celery/celery#6760]
# Checklist

- [ ] I have verified that the issue exists against the `master` branch of Celery.
- [ ] This has already been asked to the [discussion group](https://groups.google.com/forum/#!forum/celery-users) first.
- [x] I have read the relevant section in the
  [contribution guide](http://docs.celeryproject.org/en/latest/contributing.html#other-bugs)
  on reporting bugs.
- [x] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Bug+Report%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical bug reports.
- [x] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22PR+Type%3A+Bugfix%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
- [x] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the bug was already fixed in the master branch.
- [x] I have included all related issues and possible duplicate issues
  in this issue (If there are none, check this box anyway).

## Mandatory Debugging Information

- [ ] I have included the output of ``celery -A proj report`` in the issue.
    (if you are not able to do this, then at least specify the Celery
     version affected).
- [ ] I have verified that the issue exists against the `master` branch of Celery.
- [x] I have included the contents of ``pip freeze`` in the issue.
- [ ] I have included all the versions of all the external dependencies required
  to reproduce this bug.

## Optional Debugging Information

- [ ] I have tried reproducing the issue on more than one Python version
  and/or implementation.
- [ ] I have tried reproducing the issue on more than one message broker and/or
  result backend.
- [x] I have tried reproducing the issue on more than one version of the message
  broker and/or result backend.
- [ ] I have tried reproducing the issue on more than one operating system.
- [ ] I have tried reproducing the issue on more than one workers pool.
- [ ] I have tried reproducing the issue with autoscaling, retries,
  ETA/Countdown & rate limits disabled.
- [x] I have tried reproducing the issue after downgrading
  and/or upgrading Celery and its dependencies.

#### Related Issues

- Retry not working properly when using chunks #6755
- Pickling Retry instance is incorrect #6748

#### Possible Duplicates

- None

## Environment & Settings

**Celery version**: 5.0.5

<details>
<summary><b><code>celery report</code> Output:</b></summary>
<p>

```
```

</p>
</details>

# Steps to Reproduce

Create a django application with celery with default settings using pyamqp and rabbitmq 3.8.16 (current version) as broker, and run it in docker using docker-compose.

```
services:
  rabbit:
    container_name: rabbit
    image: rabbitmq:3.8.16-management
    hostname: myapp-rabbit
  celery:
    command: [
      "bash", "-c", "date && celery -A myapp worker -l debug --scheduler django_celery_beat.schedulers:DatabaseScheduler -B"
    ]
    image: <image based on ubuntu:20.04>
    container_name: celery
    depends_on:
      - rabbit
      - db
   db:
    image: postgres:10
```

 Implement a task that will call retry, and then start everything and set up the scheduler to run the task.
```
@app.task(bind=True)
def debug_task(self):
    print("Running debug task")
    self.retry(countdown=60 * 60, max_retries=1)
```

## Required Dependencies

* **Minimal Python Version**: 3.8
* **Minimal Celery Version**: 5.0.5
* **Minimal Kombu Version**: 5.0.2
* **Minimal Broker Version**: rabbit 3.8.16
* **Minimal Result Backend Version**: N/A or Unknown
* **Minimal OS and/or Kernel Version**: ubuntu 20.04
* **Minimal Broker Client Version**: N/A or Unknown
* **Minimal Result Backend Client Version**: N/A or Unknown

### Python Packages

<details>
<summary><b><code>pip freeze</code> Output:</b>
<p>

```
Django==3.1.5
amqp==5.0.6
celery==5.0.5
django-celery-beat==2.1.0
django-celery-results==2.0.1
kombu==5.0.2</summary>
```

</p>
</details>

### Other Dependencies

<details>
<p>
N/A
</p>
</details>

## Minimally Reproducible Test Case

<details>
<p>

```python
```

</p>
</details>

# Expected Behavior
Expected the task to retry after one hour

# Actual Behavior

After 15 minutes, the entire celery container crashes with the following error
```
07/05/2021 06:51:38 MainProcess    : celery.worker  : CRITICAL Unrecoverable error: PreconditionFailed(406, 'PRECONDITION_FAILED - consumer ack timed out on channel 1', (0, 0), '')
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/celery/worker/worker.py", line 203, in start
    self.blueprint.start(self)
  File "/usr/local/lib/python3.8/dist-packages/celery/bootsteps.py", line 116, in start
    step.start(parent)
  File "/usr/local/lib/python3.8/dist-packages/celery/bootsteps.py", line 365, in start
    return self.obj.start()
  File "/usr/local/lib/python3.8/dist-packages/celery/worker/consumer/consumer.py", line 311, in start
    blueprint.start(self)
  File "/usr/local/lib/python3.8/dist-packages/celery/bootsteps.py", line 116, in start
    step.start(parent)
  File "/usr/local/lib/python3.8/dist-packages/celery/worker/consumer/consumer.py", line 592, in start
    c.loop(*c.loop_args())
  File "/usr/local/lib/python3.8/dist-packages/celery/worker/loops.py", line 81, in asynloop
    next(loop)
  File "/usr/local/lib/python3.8/dist-packages/kombu/asynchronous/hub.py", line 361, in create_loop
    cb(*cbargs)
  File "/usr/local/lib/python3.8/dist-packages/kombu/transport/base.py", line 235, in on_readable
    reader(loop)
  File "/usr/local/lib/python3.8/dist-packages/kombu/transport/base.py", line 217, in _read
    drain_events(timeout=0)
  File "/usr/local/lib/python3.8/dist-packages/amqp/connection.py", line 523, in drain_events
    while not self.blocking_read(timeout):
  File "/usr/local/lib/python3.8/dist-packages/amqp/connection.py", line 529, in blocking_read
    return self.on_inbound_frame(frame)
  File "/usr/local/lib/python3.8/dist-packages/amqp/method_framing.py", line 53, in on_frame
    callback(channel, method_sig, buf, None)
  File "/usr/local/lib/python3.8/dist-packages/amqp/connection.py", line 535, in on_inbound_method
    return self.channels[channel_id].dispatch_method(
  File "/usr/local/lib/python3.8/dist-packages/amqp/abstract_channel.py", line 143, in dispatch_method
    listener(*args)
  File "/usr/local/lib/python3.8/dist-packages/amqp/channel.py", line 277, in _on_close
    raise error_for_code(
amqp.exceptions.PreconditionFailed: (0, 0): (406) PRECONDITION_FAILED - consumer ack timed out on channel 1
```

Rabbit logs the following:
`[warning] <0.10196.1> Consumer None4 on channel 1 has timed out waiting on consumer acknowledgement. Timeout used: 900000 ms`

Note that by downgrading rabbitmq to version 3.8.14 this doesn't happen anymore. (This might be a bug in rabbit instead.)

**Comment by cguethle** [comment]
Encountered this today as well. Different situation. What it boiled down to for me was a simple "time.sleep(1200)" in a celery job. This amounts to a 20min sleep, after 15 min of which the celery worker shutdowns with the OP's error. Another worker picks up the job and starts from the beginning.

Fixed by downgrading RabbitMQ container to 3.8.14. Thanks @89erik for the tip.

**Comment by olivierreberga** [comment]
Got the same issue here with Celery 4.4.7 and rabbitmq:3 and librabbitmq 1.5.2.
15 min after the start of the jobs, i get this error in my rabbit container

`2021-05-07 06:05:00.007 [warning] <0.833.0> Consumer None4 on channel 1 has timed out waiting on consumer acknowledgement. Timeout used: 900000 ms
2021-05-07 06:05:00.008 [error] <0.833.0> Channel error on connection <0.818.0> (X.X.X.X:48848 -> X.X.X.X:5672, vhost: '/', user: 'guest'), channel 1:
operation none caused a channel exception precondition_failed: consumer ack timed out on channel 1`

and just after, this error on celery side:

`[2021-05-07 06:05:00,010: CRITICAL/MainProcess] Unrecoverable error: PreconditionFailed(406, 'PRECONDITION_FAILED - consumer ack timed out on channel 1', (0, 0), '')
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/celery/worker/worker.py", line 208, in start
    self.blueprint.start(self)
  File "/usr/local/lib/python3.7/site-packages/celery/bootsteps.py", line 119, in start
    step.start(parent)
  File "/usr/local/lib/python3.7/site-packages/celery/bootsteps.py", line 369, in start
    return self.obj.start()
  File "/usr/local/lib/python3.7/site-packages/celery/worker/consumer/consumer.py", line 318, in start
    blueprint.start(self)
  File "/usr/local/lib/python3.7/site-packages/celery/bootsteps.py", line 119, in start
    step.start(parent)
  File "/usr/local/lib/python3.7/site-packages/celery/worker/consumer/consumer.py", line 599, in start
    c.loop(*c.loop_args())
  File "/usr/local/lib/python3.7/site-packages/celery/worker/loops.py", line 83, in asynloop
    next(loop)
  File "/usr/local/lib/python3.7/site-packages/kombu/asynchronous/hub.py", line 364, in create_loop
    cb(*cbargs)
  File "/usr/local/lib/python3.7/site-packages/kombu/transport/base.py", line 238, in on_readable
    reader(loop)
  File "/usr/local/lib/python3.7/site-packages/kombu/transport/base.py", line 220, in _read
    drain_events(timeout=0)
  File "/usr/local/lib/python3.7/site-packages/amqp/connection.py", line 508, in drain_events
    while not self.blocking_read(timeout):
  File "/usr/local/lib/python3.7/site-packages/amqp/connection.py", line 514, in blocking_read
    return self.on_inbound_frame(frame)
  File "/usr/local/lib/python3.7/site-packages/amqp/method_framing.py", line 55, in on_frame
    callback(channel, method_sig, buf, None)
  File "/usr/local/lib/python3.7/site-packages/amqp/connection.py", line 521, in on_inbound_method
    method_sig, payload, content,
  File "/usr/local/lib/python3.7/site-packages/amqp/abstract_channel.py", line 145, in dispatch_method
    listener(*args)
  File "/usr/local/lib/python3.7/site-packages/amqp/channel.py", line 280, in _on_close
    reply_code, reply_text, (class_id, method_id), ChannelError,
amqp.exceptions.PreconditionFailed: (0, 0): (406) PRECONDITION_FAILED - consumer ack timed out on channel 1`

**Comment by olliemath** [comment]
The same here with `Celery 4.4.7` and `rabbitmq:3.8`. An unrecoverable error caused by a long-running job on some production systems. Systems on 3.8.14 were OK, and 3.8.16 not.

**Comment by olliemath** [comment]
Possibly connected with the introduction of the `consumer_timeout` parameter and this https://github.com/rabbitmq/rabbitmq-server/issues/3032

**Comment by thedrow** [comment]
We should document this problem and the solution. Let me know if you reach a final conclusion.

**Comment by Ian-Butler-Novacoast** [comment]
This does seem related to the consumer_timeout change in Rabbit, I have observed the same error when scheduling a task with a countdown of more than 15 minutes and no other tasks being handled by the worker, increasing the `consumer_timeout` was the fix/work-around.

**Comment by thedrow** [comment]
Could one of you please contribute a documentation enhancement PR?

**Comment by olivierreberga** [comment]
I also confirm that downgrading RabbitMQ container from 3.8.16 to 3.8.14 has solved the issue. Thanks @89erik for the tip.

**Comment by coweos** [comment]
I confirm that downgrading RabbitMQ from 3.8.16 to 3.8.14 has solved the issue on my side too

**Comment by tahamr83** [comment]
I confirm that downgrading RabbitMQ from 3.9.* to 3.8.16 solved the issue for us.

**Comment by auvipy** [comment]
i can see several related rabbitmq issue like this

**Comment by auvipy** [comment]
@alphavector

**Comment by alphavector** [comment]
**Short answer:**
1) Use a version of `rabbitmq < 3.8.15`
OR
2) If using `rabbitmq >= 3.8.15`, set `consumer_timeout = 31622400000` in `rabbitmq.conf`

If you are using official [rabbitmq images](https://hub.docker.com/_/rabbitmq), the configuration file might look something like this.

**rabbitmq.conf**
```
loopback_users.guest = false
listeners.tcp.default = 5672
consumer_timeout = 31622400000 # 1 year or any other number in milliseconds
```
**docker-compose.yml**
```
services:
  rabbitmq:
    image: rabbitmq:3.8.21
    volumes:
      - "./rabbitmq.conf:/etc/rabbitmq/rabbitmq.conf"
```


**Detailed answer:**
With this pr https://github.com/rabbitmq/rabbitmq-server/pull/2990, the contributors set the default value for consumer_timeout to `15 minutes` in `3.8.15`, then increased it to `30 minutes` in `3.8.17` https://github.com/rabbitmq/rabbitmq-server/pull/3033 

This change has also been documented recently: https://www.rabbitmq.com/consumers.html#acknowledgement-timeout 

If the consumer does not acknowledge the message within this time, `rabbitmq` considers the consumer to be faulty and closes the channel.

I suppose when we schedule a task far in the future, message acknowledgement is not sent before the time specified in `eta (countdown)` and `rabbitmq` closes the connection.

Specifying `consumer_timeout = 31622400000` (1 year in milliseconds) gives a 99.9% guarantee that no problem will occur in the future. Unfortunately, you can't specify `false` or `undefined`, because an integer is expected (https://github.com/rabbitmq/rabbitmq-server/blob/13ac078746c0f19635562f591a0b7e03b8a87c0c/deps/rabbit/priv/schema/rabbit.schema#L1291).

I can't say that if `consumer_timeout < countdown`, the tasks will stop working. I've set consumer_timeout to 5 milliseconds and countdown to 20 seconds - the tasks run without errors, although the difference is 4000 times (pretty weird...)

Question for discussion, do we document all of the above, or do we change the mechanism of delayed tasks. @celery/core-developers @auvipy, I need your opinion.

**Comment by auvipy** [comment]
i think  change the mechanism of delayed tasks is the best answer, but we still need to document the reasoning of the change :)

**Comment by thedrow** [comment]
@auvipy How would you propose that we change how we implement tasks with ETAs?

The current design has the following properties:

1. We consume the message and wait until we're past the ETA and execute it.
2. If the consumer times out, we return the message to the queue without acknowledging it and therefore it will be consumed by another worker in the future.

I don't see anything flawed with this design nor do I see another way to implement the same feature.

However, if the consumer times out, RabbitMQ closes the channel.
This forces a worker to restart, unfortunately.
We should catch that exception and restart the worker.

If we do restart for that reason, we should log why the restart occurred.
In addition, we need to document everything @alphavector mentioed.

**Comment by thedrow** [comment]
This is still not done. We need to recover from this error.

**Comment by alphavector** [comment]
@auvipy, yes, we have documented the problem, now we need to make worker restart automatically when such an error occurs. @thedrow, if you don't mind, I could try to take over this task

**Comment by auvipy** [comment]
> This is still not done. We need to recover from this error.

sorry. forgot about that :D

**Comment by auvipy** [comment]
> @auvipy, yes, we have documented the problem, now we need to make worker restart automatically when such an error occurs. @thedrow, if you don't mind, I could try to take over this task

please go for it!

**Comment by lowercase00** [comment]
I've been having a similar issue after I changed to `gevent` with `concurrency=5` and increased the countdown on the retry logic (currently using `retry * 120`), but don't know if it's related. RabbitMQ 3.8.23, but didn't have this issue with the same RabbitMQ version using prefork.

The exact error I got is `PRECONDITION_FAILED - delivery acknowledgement on channel 1 timed out. Timeout value used: 1800000 ms. This timeout value can be configured, see consumers doc guide to learn more`.

It seems that this only happens when tasks are retried. I'm not sure whats the mechanism of a retried task, but it seems that it stays "live" on the queue and on the worker, instead of getting back into the queue at a later time, as if the broker was still expecting for acknowledgement, if though the task is on hold (no acknowledgement needed)? Not sure if that makes sense.

**Comment by nodesocket** [comment]
Is there away to detect this condition? We can just have Kubernetes `livenessProbe` detect this in our celery workers and then it will fail the healthcheck and Kubernetes will gracefully restart the broken celery pods.

**Comment by eljeffeg** [comment]
Just to clarify, is the celery consumer suppose to auto ack?  I guess I want to know if perhaps I'm not properly acknowledging the task after I do a `celery.send_task`.  I store the task and then look through it and check the status, but I don't call an `ack` on it.

I expect this situation would also occur if I restarted the consumer docker and some task was left hanging out there?  I've increased `consumer_timeout` a bit, but it sort of feels like cutting the dashboard lights.

**Comment by glormph** [comment]
Also when running a pretty default setup (no countdown, no ETA, no `acks_late`) this issue bit me in the ass, with tasks that run > `consumer_timeout`. But only when the number of tasks was above the concurrency.

I guess in my case the prefetched tasks are not acked and then the timeout hits? I tried running with `acks_late` and `worker_prefetch_multiplier = 1` there is no prefetching but then of course the task to execute is not acked within the `consumer_timeout` and the problem is still there.

Setting

**Comment by christopher5106** [comment]
I got this error when I started  to increase concurrency parameter of the worker from 1 to 8. It seams to me that the Celery worker.py does not recover from this error. Celery code should be updated, because at my level of tasks, I can't do anything, I believe.

**Comment by darixsamani** [comment]
I have this error in celery-worker

```
celery-worker_1  | [2024-04-27 08:33:21,580: CRITICAL/MainProcess] Unrecoverable error: PreconditionFailed(406, 'PRECONDITION_FAILED - delivery acknowledgement on channel 1 timed out. Timeout value used: 1800000 ms. This timeout value can be configured, see consumers doc guide to learn more', (0, 0), '')
celery-worker_1  | Traceback (most recent call last):
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/celery/worker/worker.py", line 202, in start
celery-worker_1  |     self.blueprint.start(self)
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/celery/bootsteps.py", line 116, in start
celery-worker_1  |     step.start(parent)
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/celery/bootsteps.py", line 365, in start
celery-worker_1  |     return self.obj.start()
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/celery/worker/consumer/consumer.py", line 340, in start
celery-worker_1  |     blueprint.start(self)
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/celery/bootsteps.py", line 116, in start
celery-worker_1  |     step.start(parent)
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/celery/worker/consumer/consumer.py", line 742, in start
celery-worker_1  |     c.loop(*c.loop_args())
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/celery/worker/loops.py", line 97, in asynloop
celery-worker_1  |     next(loop)
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/kombu/asynchronous/hub.py", line 373, in create_loop
celery-worker_1  |     cb(*cbargs)
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/kombu/transport/base.py", line 248, in on_readable
celery-worker_1  |     reader(loop)
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/kombu/transport/base.py", line 230, in _read
celery-worker_1  |     drain_events(timeout=0)
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/amqp/connection.py", line 526, in drain_events
celery-worker_1  |     while not self.blocking_read(timeout):
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/amqp/connection.py", line 532, in blocking_read
celery-worker_1  |     return self.on_inbound_frame(frame)
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/amqp/method_framing.py", line 53, in on_frame
celery-worker_1  |     callback(channel, method_sig, buf, None)
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/amqp/connection.py", line 538, in on_inbound_method
celery-worker_1  |     return self.channels[channel_id].dispatch_method(
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/amqp/abstract_channel.py", line 156, in dispatch_method
celery-worker_1  |     listener(*args)
celery-worker_1  |   File "/usr/local/lib/python3.10/site-packages/amqp/channel.py", line 293, in _on_close
celery-worker_1  |     raise error_for_code(
celery-worker_1  | amqp.exceptions.PreconditionFailed: (0, 0): (406) PRECONDITION_FAILED - delivery acknowledgement on channel 1 timed out. Timeout value used: 1800000 ms. This timeout value can be configured, see consumers doc guide to learn more

```

How can I solve the error where the worker shuts down after a certain number of minutes?

**Comment by dineshtrivedi** [comment]
Hello guys, I was planning to use the `consumer_timeout` configuration in RabbitMQ similar to `visbility_timeout` with Redis to re-enqueue tasks not too far in the future in case a killed or faulty worker. However, I am experiencing the same error as above. Is there any suggestion on how to tackle this?

Thanks a lot

**Comment by riccardomassullo** [comment]
For us, the broker option [broker_channel_error_retry](https://docs.celeryq.dev/en/stable/userguide/configuration.html#broker-channel-error-retry) provided by Celery is doing the trick (the trick is to reconnect and give a chance to finish and ack the job):

```
amqp.exceptions.PreconditionFailed: (0, 0): (406) PRECONDITION_FAILED - delivery acknowledgement on channel 1 timed out. Timeout value used: 1800000 ms. This timeout value can be configured, see consumers doc guide to learn more
[2024-06-19 13:46:42,971: WARNING/MainProcess] /api/.local/lib/python3.8/site-packages/celery/worker/consumer/consumer.py:391: CPendingDeprecationWarning: 
In Celery 5.1 we introduced an optional breaking change which
on connection loss cancels all currently executed tasks with late acknowledgement enabled.
These tasks cannot be acknowledged as the connection is gone, and the tasks are automatically redelivered
back to the queue. You can enable this behavior using the worker_cancel_long_running_tasks_on_connection_loss
setting. In Celery 5.1 it is set to False by default. The setting will be set to True by default in Celery 6.0.

  warnings.warn(CANCEL_TASKS_BY_DEFAULT, CPendingDeprecationWarning)

[2024-06-19 13:46:42,983: INFO/MainProcess] Connected to amqp://****:****@****:5672//
[2024-06-19 13:46:43,007: INFO/MainProcess] mingle: searching for neighbors
[2024-06-19 13:46:44,114: INFO/MainProcess] mingle: sync with 61 nodes
[2024-06-19 13:46:44,114: INFO/MainProcess] mingle: sync complete
[2024-06-19 13:46:44,192: INFO/MainProcess] Task sherpa.celery_worker.main.handle[6d2a6ad3-e38a-4107-a35a-f11d79fdaad8] received
[2024-06-19 13:46:44,193: INFO/MainProcess] Task sherpa.celery_worker.main.handle[dae1bdac-f401-4771-8b55-0a03cb22378a] received
[2024-06-19 13:46:44,194: INFO/MainProcess] Task sherpa.celery_worker.main.handle[b7408c27-a730-490a-8a62-0b6530288d0c] received
[2024-06-19 13:46:44,195: INFO/MainProcess] Task sherpa.celery_worker.main.handle[a906db84-def8-4309-9d93-94b26c095b8e] received
[2024-06-19 13:47:44,242: INFO/ForkPoolWorker-1] Task sherpa.celery_worker.main.handle[6d2a6ad3-e38a-4107-a35a-f11d79fdaad8] succeeded in 1861.319063273986s: ...
```

**Comment by dineshtrivedi** [comment]
Thank you, @riccardomassullo!
We are using the configurations below and it is also working now.
```
CELERY_BROKER_CONNECTION_RETRY = True
CELERY_BROKER_CHANNEL_ERROR_RETRY = True
CELERY_WORKER_CANCEL_LONG_RUNNING_TASKS_ON_CONNECTION_LOSS = True
CELERY_BROKER_CONNECTION_RETRY_ON_STARTUP = True
```

**Comment by newbie3033** [comment]
Hi all, I ran into the same error, but the above solutions don't seem to handle the following case:

- both of `worker_prefetch_multiplier ` and `worker_concurrency `  in celery set to 1

- set `broker_channel_error_retry` to True

- `consumer_timeout ` in rabbitmq.conf set to 1000

- just start a single worker and send three task messages to it(each task took well over 1,000 milliseconds to run) 

Then it happens:  the worker repeatedly receives and executes the last two tasks, and keeps repeating the last one task ( we can see that there is always an unack message in the rabbitmq queue).

I looked for configuration parameters for this :
 `worker_cancel_long_running_tasks_on_connection_loss` 
`worker_enable_prefetch_count_reduction` 
`worker_deduplicate_successful_tasks` 
,but none of them seem to solve the dilemma.

Is there a solution other than setting the value of `consumer_timeout` as large as possible?
